Add a temporary `--benchmarks` flag for internal compatibility

oontvoo opened the pull request:
Details:

Internally, we use `--benchmarks` flag to specify which benchmarks to run. In the OSS version here, however, there is only `--benchmark_filter` (which does the same thing).
This patch added the alias so that we could gradually migrate users over to using `--benchmark_filter`

oontvoo commented:
@dominichamon

LebedevRI commented:
This is missing some words.

oontvoo commented:
> This is missing some words.

Thanks - it got truncated because the sentence was too long. I've updated it to be shorter

LebedevRI commented:
> > This is missing some words.
> 
> Thanks - it got truncated because the sentence was too long. I've updated it to be shorter

I was mainly talking about the PR description

oontvoo commented:
> > > This is missing some words.
> > 
> > 
> > Thanks - it got truncated because the sentence was too long. I've updated it to be shorter
> 
> I was mainly talking about the PR description

Updated the first comment to have more details.

pleroy commented:
This PR looks like a terrible idea to me.  There is no way that the flag can ever be removed once it's added, because any user of the library is free to use it instead of `--benchmark_filter`.  The `FIXME` is not going to help (and it won't be fixed for a long time anyway).

I surely hope that there is a way to deal with internal compatibility without polluting the public API.

dominichamon commented:
i suggested this because we only ever update the internal version from the external version, and any internal-only changes would be overwritten every time.

however, @pleroy, you're absolutely right and maybe we need to just handle this internally as we migrate, and trust our tooling to identify this issue when imports happen.

pleroy commented:
Maybe you could have a small file with a module initializer that copies `--benchmarks` to `--benchmark_filter` if it is set, and include that at link time in the `BUILD` file.

oontvoo commented:
Thanks - closing this PR now. No longer needed as we'll go with a different approach
